### PR TITLE
Feature/adopty

### PR DESCRIPTION
**Pull request by MaksimLihoi**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -35,6 +40,28 @@ const App = () => {
     init(AMPLITUDE_API_KEY);
   }, []);
 
+  useEffect(() => {
+    activateAdapty({ sdkKey: 'public_live_fabSVYF8.r7jC2ljNCMWNTyjOy0RR' });
+  }, []);
+
+  useEffect(() => {
```
Comment (kirillkurko) [inline App.js:47]: What if we create custom to encapsulate this logic? Maybe we can have something `useAdapty` (the name could be improved) where you set the SDK key and execute the logic from that `useEffect`

File: src/screens/Onboarding/NameStep/index.js [hunk 2]
```diff
@@ -46,13 +51,44 @@ const NameStep = (props) => {
   const navigation = useNavigation();
   const track = useAnalytics();
 
+  const [trackingStatus, setTrackingStatus] = React.useState<
```
Comment (kirillkurko) [inline src/screens/Onboarding/NameStep/index.js:54]:
```suggestion
  const [trackingStatus, setTrackingStatus] = useState<
```

File: src/screens/Onboarding/NameStep/index.js [hunk 3]
```diff
@@ -46,13 +51,44 @@ const NameStep = (props) => {
...
+  >('(loading)');
+
   const [state, setState] = useState({
     value: '',
     shift: new Animated.Value(0),
     isAnimate: true,
     placeholder: resources.t('ONBOARDING.PLACEHOLDER'),
   });
 
+  const request = React.useCallback(async () => {
```
Comment (kirillkurko) [inline src/screens/Onboarding/NameStep/index.js:65]: Let's import `useCallback`, so you can use as `useCallback(...)` and not as `React.useCallback(...)`

File: App.js [hunk 4]
```diff
@@ -35,6 +40,28 @@ const App = () => {
...
+          if (error) {
+            if (__DEV__) {
+              console.error(error);
+            }
+          } else {
+            adapty.updateAttribution(uid, installData, 'AppsFlyer');
+          }
+        });
+      } catch (error) {
+        console.log(error);
```
Comment (kirillkurko) [inline App.js:60]: According to other logs in this pull request it should be implemented as in the snippet below:
```
if (__DEV__) {
  console.error(error);
}
```

File: src/screens/Onboarding/NameStep/index.js [hunk 5]
```diff
@@ -46,13 +51,44 @@ const NameStep = (props) => {
...
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    request();
+    getTrackingStatus()
+      .then((status) => {
+        setTrackingStatus(status);
+        if (__DEV__) {
```
Comment (kirillkurko) [inline src/screens/Onboarding/NameStep/index.js:81]: What if you'll implement your own simple logger just to hide this if (__DEV__) condition: 

```
// logs only if __DEV__ is equals to true 
logger.log(message) // console.log
logger.error(message) console.error
```

**Review by kirillkurko — COMMENTED** [review]
Nothing criminal - looks good to me. Left some minor comments and suggested some possible improvements

**Review by kirillkurko — APPROVED** [review]
LGTM 🚀

File: src/screens/Onboarding/NameStep/index.js (hunk 5 above)
Comment (MaksimLihoi) [reply src/screens/Onboarding/NameStep/index.js:81]: done

File: App.js (hunk 4 above)
Comment (MaksimLihoi) [reply App.js:60]: done

**Pull request merged**